rs/decentralization/src/network.rs

```diff
@@ -758,18 +768,34 @@ pub trait TopologyManager: SubnetQuerier + AvailableNodesQuerier {
             min_nakamoto_coefficients,
             ..Default::default()
         }
-        .with_include_nodes(include_nodes.clone())
-        .with_exclude_nodes(exclude_nodes.clone())
-        .with_only_nodes_that_have_features(only_nodes.clone())
+        .including_from_available(NodeSelector::PrincipalIdList(include_nodes.clone()))
+        .excluding_from_available(NodeSelector::FeatureList(exclude_nodes.clone()))
+        .including_from_available(NodeSelector::FeatureList(only_nodes.clone()))
         .resize(size, 0)
     }
 }
 
+pub enum NodeSelector {
```

Comment:
I'm struggling to understand how `NodeSelector` works, and that's not a good sign.
Could you add some comments to make it easier for me? Please? :D

Reply:
I added some examples and comments ;)